common/thorhelper/hpccconfig.cpp

```diff
@@ -88,7 +90,26 @@ void getMemorySpecifications(std::unordered_map<std::string, __uint64> &memorySp
                 WARNLOG("The total memory requirements of the query (%u MB) exceed the recommended limits for '%s' (total memory: %u MB, recommended max percentage : %.2f%%)", (unsigned)(totalRequirements / 0x100000), context, maxMB, maxPercentage);
         }
     }
-    memorySpecifications["recommendedMaxMemory"] = recommendedMaxMemory;
+    memorySpecifications->setPropInt64("@recommendedMaxMemory", recommendedMaxMemory);
+    memorySpecifications->setPropInt64("@totalMemory", maxBytes);
+
+    // a simple helper used below, to fetch bool from workunit, or the memory settings (either managerMemory or workerMemory) or legacy location
```

Comment:
Looks like this comment it out of date, shouldn't specifically refer to manager/worker..